Add Glove80 keypos definitions

hylophile opened the pull request:
Hi,

first off, thank you for this project!
This PR adds key position definitions for the Glove80 keyboard.

Comparing with the [ADV360 thumb key definitions](https://github.com/urob/zmk-nodefree-config/blob/main/keypos_def/keypos_adv360.h#L10), my choice here is probably not really compatible for someone using both boards, since for the ADV360, the two easiest to reach thumb keys are XH0 and XH1, whereas here they're XH1 and XH2. On the other hand, it'd be confusing for users who only use the Glove80 to use XH1 XH0 XH2. Not sure what's the best approach here.

I also wrote a macro for defining Glove80 layers which is quite handy. The key positions for the two thumb rows are between rows (and not after them), which ends up taking a lot of vertical space when aligning them properly. The macro just takes a couple of arguments and rearranges them to fix this, allowing users to define layouts the following way:
```c
GLV80_LAYER(template,
/* ╭───────────────┬───────────────┬───────────────┬───────────────┬───────────────╮                                   ╭───────────────┬───────────────┬───────────────┬───────────────┬───────────────╮ */
    ___             ___             ___             ___             ___                                                 ___             ___             ___             ___             ___
/* ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────╮   ╭───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤ */
    ___             ___             ___             ___             ___             ___                 ___             ___             ___             ___             ___             ___
/* ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤   ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤ */
    ___             ___             ___             ___             ___             ___                 ___             ___             ___             ___             ___             ___
/* ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤   ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤ */
    ___             ___             ___             ___             ___             ___                 ___             ___             ___             ___             ___             ___
/* ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤   ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤ */
    ___             ___             ___             ___             ___             ___              ,  ___             ___             ___             ___             ___             ___
/* ├───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────╯   ╰───────────────┼───────────────┼───────────────┼───────────────┼───────────────┼───────────────┤ */
    ___             ___             ___             ___             ___                              ,                  ___             ___             ___             ___             ___
/* ╰───────────────┴───────────────┴───────────────┴───────────────┴───────────────╯                                   ╰───────────────┴───────────────┴───────────────┴───────────────┴───────────────╯ */
                                                                                                     ,
/*                                         ╭───────────────┬───────────────┬───────────────╮                   ╭───────────────┬───────────────┬───────────────╮                                         */
                                            ___             ___             ___                                 ___             ___             ___
/*                                         ├───────────────┼───────────────┼───────────────┤      */ , /*      ├───────────────┼───────────────┼───────────────┤                                         */
                                            ___             ___             ___                                 ___             ___             ___
/*                                         ╰───────────────┴───────────────┴───────────────╯                   ╰───────────────┴───────────────┴───────────────╯                                         */
)
```

I'd be glad to include the macro in this PR, but wasn't sure whether it fits the project. In any case, I'll leave it here for posterity.

urob commented:
Thanks! Looks good to me. The layer macro seems useful and it doesn't hurt to include it in the definition file. So if you want to update the PR, I'd be happy to include it.

hylophile commented:
Great! I've added the macro. Thanks for the quick feedback!

hylophile commented:
Done. I've also changed the numbering slightly so that LFx and LCx align with the other rows, and renamed the macro to `GLV80_LAYER`. The first comment now includes a proper template for users.

urob commented:
Merged! Thanks for submitting!

BlueDrink9 commented:
I'm just adding a note here because I could not figure out how to use this template. I miss that there were uncommented lines between the shells. [This repo](https://github.com/adamwalz/glove80-zmk-config/blob/62fc45917023505a9fb63959069f0afd5386d67d/config/glove80.keymap#L152) is a good example of how to use it
